api-extractor/src/JsonFile.ts

```diff
@@ -69,7 +69,9 @@ export default class JsonFile {
         // tslint:disable-next-line:no-any
         const value: any = json[key];
         if (value === undefined) {
-          throw new Error(`The key "${key}" is undefined`);
+          if (key !== 'type') {
```

Comment:
Why are we trying to write "undefined" values into a  JSON file?  IIRC JSON doesn't support undefined.